client/src/main/java/io/confluent/kafka/schemaregistry/client/MockSchemaRegistryClient.java

```diff
@@ -273,4 +274,35 @@ public String getCompatibility(String subject) throws IOException, RestClientExc
   public int getId(String subject, Schema schema) throws IOException, RestClientException {
     return getIdFromRegistry(subject, schema, false);
   }
+
+  @Override
+  public List<Integer> deleteSubject(String subject) throws IOException, RestClientException {
+    return deleteSubject(null, subject);
+  }
+
+  @Override
+  public List<Integer> deleteSubject(Map<String, String> requestProperties, String subject)
+      throws IOException, RestClientException {
+    schemaCache.remove(subject);
+    idCache.remove(subject);
```

Comment:
we can't remove from id cache. The functionality is such that we will only remove the subject version but not the underlying id. This is is to provide protection against the old records stha could potentially still be not consumed.

Reply:
Removed this line.